src/main/java/de/azapps/kafkabackup/common/record/RecordJSONSerde.java

```diff
@@ -54,14 +57,27 @@ public Record deserialize(JsonParser jp, DeserializationContext ctx) throws IOEx
             String topic = node.get(TOPIC_PROPERTY).asText();
             int partition = node.get(PARTITION_PROPERTY).asInt();
             long offset = node.get(OFFSET_PROPERTY).asLong();
+            String timestampTypeStr = node.get(TIMESTAMP_TYPE_PROPERTY).asText(null); // Default seems to be the string "null", which is not wanted here
+            TimestampType timestampType;
+            if (timestampTypeStr != null) {
+                timestampType = TimestampType.forName(timestampTypeStr);
+            } else {
+                timestampType = TimestampType.NO_TIMESTAMP_TYPE;
+            }
```

Comment:
maybe: `TimestampType timestampType = Optional.ofNullable(node.get(TIMESTAMP_TYPE_PROPERTY).asText(null))
.map(timestampTypeStr -> TimestampType.forName(timestampTypeStr))
.orElseGet(TimestampType.NO_TIMESTAMP_TYPE)` ?

Reply:
Java is getting functional. Love it!